src/auth/digest/Nonce.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 1996-2025 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_AUTH_DIGEST_NONCE_H
+#define SQUID_SRC_AUTH_DIGEST_NONCE_H
+
+#include "auth/digest/forward.h"
+#include "hash.h"
+#include "mem/AllocatorProxy.h"
+
+namespace Auth
+{
+namespace Digest
+{
+
+class Nonce : public hash_link
+{
+    MEMPROXY_CLASS(Auth::Digest::Nonce);
+public:
+    /// data to be encoded into the nonce's hex representation
+    struct _Data {
```

Comment:
If we do not have to start this name with an underscore, let's not do that:

```suggestion
    struct Data {
```

or even completely remove that name:

```suggestion
    struct {
```